### DISCUSSION TITLE
Misunderstanding of the rf neuron.frequency / period.

### DISCUSSION THREAD

**Discussion by zeinebCh** [lava-nc/lava-dl#178] [General]
Hello everyone,

I have a misunderstanding of the neuron.frequency: Is it the sampling frequency or the resonant frequency of the neuron?

In my case I have a starting frequency of 58 GHz, End frequency of 63.5 GHz, Bandwidth of 5.5 GHz. I have a layer consisting of 64 Neuron. I want to assign a different resonant frequency to each of our 64 samples (kind of mimicking FFTs) to cover the whole 5.5 GHz. Would this be achievable via LAVA, how about the period/frequency definition in such a case?

I would appreciate a lot your help. 
Thank you

**Comment by bamsumit** [comment]
@zeinebCh it is the resonant frequency of the neuron. The frequency/period is in discrete-time units and is dependent on your sampling frequency (1 / time-step).

For your 63.5 GHz signal, the $f_s > 127 \text{GHz} \implies \text{time-step} <  7.8\text{ps}$ which is unrealistic for real-time processing. You can do it with small enough time-step if you are not concerned about real-time processing.
